Fix few docs tickets (#2793)
mattermost/docs: source/administration/config-settings.rst

When using High Availability, the salt must be identical in each instance of Mattermost.

The following fields are encrypted using this key

- ``SqlSettings.DriverName``
- ``SqlSettings.DataSource``
- ``SqlSettings.MaxIdleConns``
- ``SqlSettings.MaxOpenConns``
- ``SqlSettings.Trace``
- ``SqlSettings.QueryTimeout``

> - ``SqlSettings.ConnMaxLifetimeMilliseconds``

Reviewer: I'm pretty sure this isn't true. I don't think the AtRestEncrypt key is used for anything at the moment and is a legacy setting. I'm curious as to how you came up with that?

Author: I found `AtRestEncrypt` https://github.com/search?p=2&q=org%3Amattermost+SqlSettings&type=Code being set at quite a few places, I think the fields of `SqlSettings` are being encrypted using `AtRestEncrypt`, right?

Author: See this https://github.com/search?q=org%3Amattermost+SqlSettings.AtRestEncryptKey&type=Code

Reviewer: Sure there are reference to it in code but it's not being used for any actual encrypting. Can you point to where it's actually doing what you say?

>The following fields are encrypted using this key

Author: I looked at quite a lot of things, I don't remember now, correct if I'm wrong.
- https://github.com/mattermost/mattermost-server/commit/34d688ca72d8ef13b83987cc3b3d9b93ab989d9f
- https://github.com/mattermost/docs/issues/2231
- https://github.com/mattermost/mattermost-server/issues/9741
- https://github.com/mattermost/mattermost-redhat-docker/blob/e9d707231cc867fcc265fdb3ebe4073c3e91a340/v5/entrypoint.sh#L52

I didn't find a direct reference of `AtRestEncryptKey`, what I understood was `AtRestEncryptKey` encrypts anything `SqlSettings`. I agree that there is no need for encrypting all fields of `SqlSettings`. Why do you have `AtRestEncryptKey` then? Just curious.

Reviewer: It's a legacy setting that we used to use for encrypting data being stored at rest in the database. The setting was never really fully used and we just haven't removed it

Reviewer: If you want to remove this section from the PR the other parts look good

Author: I added a31109e, looks ok? or should I remove this line? I suggest you create a ticket for this.

Author: Does a31109e, looks ok? or should I remove the line? I suggest you create a ticket.

Reviewer: That's fine, thanks!

When using High Availability, the salt must be identical in each instance of Mattermost.

> No fields are encrypted using `AtRestEncryptKey`. It's a legacy setting used to

Reviewer 2: ```suggestion
No fields are encrypted using ``AtRestEncryptKey``. It's a legacy setting used to
```

> encrypt data being stored at rest in the database.

Reviewer 2: ```suggestion
encrypt data stored at rest in the database.
```

> use for encrypting data being stored at rest in the database.

Reviewer: ```suggestion
encrypt data being stored at rest in the database.
```

Session length for SSO authentication (days)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This setting defines the session length for SSO authentication, such as SAML, GitLab and OAuth 2.0.

Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML, GitLab or OAuth 2.0, the user may automatically be logged back in to Mattermost if they are already logged in to SAML, GitLab or with OAuth 2.0.

After changing this setting, the setting will take effect after the next time the user enters their credentials.

> +----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

Reviewer 2: Please help fix formatting of the table.

| This feature's ``config.json`` setting is ``"SessionLengthSSOInDays" : 30`` with whole number input.                                                                 |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------+

> Session length for SSO authentication (days)

Reviewer 2: I think this setting is accidentally repeated twice here.

^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> This setting defines the session length for SSO authentication, such as GitLab and SAML.

Reviewer: ```suggestion
This setting defines the session length for SSO authentication, such as SAML, GitLab and OAuth 2.0.
```

>  This setting defines the session length for SSO authentication, such as GitLab and SAML.

Reviewer: ```suggestion
This setting defines the session length for SSO authentication, such as GitLab and SAML.
```

>  Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML or GitLab, the user may automatically be logged back in to Mattermost if they are already logged in to SAML or GitLab.

Reviewer: ```suggestion
Set the number of days from the last time a user entered their credentials to the expiry of the user's session. If the authentication method is SAML, GitLab or OAuth 2.0, the user may automatically be logged back in to Mattermost if they are already logged in to SAML, GitLab or with OAuth 2.0.
```

>  After changing this setting, the setting will take effect after the next time the user enters their credentials.

Reviewer: ```suggestion
After changing this setting, the setting will take effect after the next time the user enters their credentials.
```

| This feature's ``config.json`` setting is ``"LoginButtonColor": ""`` with string input.                                       |
+-------------------------------------------------------------------------------------------------------------------------------+

SAML session lengths
^^^^^^^^^^^^^^^^^^^^^^

SAML session length is defined per

 - ``url`` : ``environment/session_lengths``
- ``title`` : ``admin.sidebar.sessionLengths``
- ``type`` : ``TYPE_NUMBER``
- ``label`` : ``mobileSessionDays``
- ``help_text`` : ``admin.service.ssoSessionDaysDesc``

> - ``placeholder``: ``sessionIdleTimeout``

Reviewer: I'm not sure the above makes sense? How does that relate to SAML session length

Author: We had a long discussion and multiple iteration on https://github.com/mattermost/docs/pull/2732, should I remove `placeholder`?

Reviewer: The content of this section doesn't make any sense to me. I'm not sure what it's trying to communicate. What does any of it mean? For example, what does the below communicate to a reader of this document?

```
``help_text`` : ``admin.service.ssoSessionDaysDesc``
```

Author: The point is, I don't remember now. I have deleted these lines - fa3495e.
